### ISSUE TITLE
TreeList should not have child that matches div[tabindex]

### ISSUE THREAD

**Issue by HazzMan2409** [telerik/kendo-angular#4852]
### Description

Updating to Kendo v22, we now get an Axe a11y violation for the TreeList component.
The `[role="treegrid"]` element has a child that matches `div[tabindex]`.

I believe this is because only elements with certain roles inside the `[role="treegrid"]` element are allowed to be focusable.

### Steps To Reproduce

Render a TreeGrid.

### Actual Behavior

The `[role="treegrid"]` element has a child that matches `div[tabindex]`.

### Expected Behavior

The `[role="treegrid"]` element should not have a child that matches `div[tabindex]`.

### Browser

Chrome

### Browser version

latest

### OS type

MacOS

**Comment by dtopalov** [comment]
Hello @HazzMan2409,

We are aware of the reported problem, but unfortunately there is little we can currently do about it. Due to [recent changes in Chrome](https://developer.chrome.com/blog/keyboard-focusable-scrollers) that (unnecessarily) forced focusing otherwise non-focusable scrollable elements (like said DIV), we had to fix the following issue - https://github.com/telerik/kendo-angular/issues/4801.

Our Grid and TreeList are complex components with multiple functionalities and specific rendering to accommodate all various states, functionalities and layouts. We have always did our best to fully comply with all WCAG criteria and accessibility guidelines. The components are fully navigable through the keyboard, and the content can be traversed and scrolled without the need of focusing the scrollable container, furthermore focusing it breaks the Grid/TreeGrid patterns that mandate the table elements to be a single tab-stop and navigation through cells to be performed using arrow keys:

https://www.w3.org/WAI/ARIA/apg/patterns/grid/#datagridsforpresentingtabularinformation

The suggested and most straight-forward approach for excluding an element from the tab sequence, is setting tabindex "-1". This on the other hand makes the element "programmatically focusable" and thus reported by AXE as a violation (whether this is a false positive is still debated in the Deque AXE tool issue tracker), but in this case we believe it is - the Grid and TreeList components are fully accessible and navigable, and the accessibility tree for assistive technologies is constructed correctly, e.g.:

<img width="1296" height="839" alt="Image" src="https://github.com/user-attachments/assets/757dc5c8-1a86-4d07-ac35-9f26c61e642a" />

Thus, the discussed warning can be safely ignored, and for the time being, we are closing this issue as "Won't Fix". We will have the case in mind when considering potential future rendering changes in the Grid and TreeList that could possibly allow us to eliminate the problem without setting tabindex to the scrollable container, but this is not yet part of our roadmap and near-future plans.

Thank you for the understanding.
